pkg/codec/www-form-urlencoded/form.go

```diff
@@ -153,5 +155,97 @@ func array(encoder url.Values, root string, refs references.Store, prop *specs.P
 // Unmarshal unmarshals the given www-form-urlencoded io reader into the given reference store.
 // This method is called during runtime to decode a new message and store it inside the given reference store
 func (manager *Manager) Unmarshal(reader io.Reader, refs references.Store) error {
+	if manager.specs == nil {
+		return nil
+	}
+
+	bb, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return err
+	}
+
+	if len(bb) == 0 {
+		return nil
+	}
+
+	values, err := url.ParseQuery(string(bb))
+	if err != nil {
+		return err
+	}
+
+	for key, values := range values {
+		if err := decodeElement(manager.resource, 0, strings.Split(key, "."), values, manager.specs.Nested, refs); err != nil {
```

Comment:
`manager.specs.Nested` could be a `nil` value. I think it is better to pass `manager.specs` directly instead.

Reply:
not sure it is possible since this func is called recursively
(which means I have to wrap that argument for the nested calls)